fedora_messaging/tests/unit/test_config.py

```diff
@@ -88,6 +88,97 @@ class = "logging.StreamHandler"
 malformed_config = 'publish_exchange = "special_exchange'  # missing close quote
 
 
+class ValidateBindingsTests(unittest.TestCase):
+    """Unit tests for :func:`fedora_messaging.config.validate_bindings`."""
+
+    def test_valid(self):
+        """Assert no exceptions are raised if the bindings are valid."""
+        bindings = [
+            {"queue": "q1", "exchange": "e1", "routing_keys": ["#"]},
+            {"queue": "q2", "exchange": "e2", "routing_keys": ("#",)},
+        ]
+
+        msg_config.validate_bindings(bindings)
+
+    def test_wrong_type(self):
+        """Assert a useful message is provided if bindings isn't a list or tuple"""
+        with self.assertRaises(ConfigurationException) as cm:
+            msg_config.validate_bindings({})
+        self.assertIn(
+            "Configuration error: bindings must be a list or tuple of dictionaries",
+            str(cm.exception),
+        )
+
+    def test_missing_keys(self):
+        """Assert a useful message is provided if "queue" is missing from the config."""
+        bindings = [{}]
+        with self.assertRaises(ConfigurationException) as cm:
+            msg_config.validate_bindings(bindings)
+        self.assertIn(
+            "Configuration error: a binding is missing the following keys",
+            str(cm.exception),
+        )
+        self.assertIn("queue", str(cm.exception))
+        self.assertIn("exchange", str(cm.exception))
+        self.assertIn("routing_keys", str(cm.exception))
+
+    def test_routing_key_str(self):
+        """Assert a useful message is provided if "routing_keys" is not a list or tuple."""
+        bindings = [{"exchange": "e1", "queue": "q1", "routing_keys": "#"}]
+        with self.assertRaises(ConfigurationException) as cm:
+            msg_config.validate_bindings(bindings)
+        self.assertIn(
+            "Configuration error: routing_keys must be a list or tuple, but was a ",
```

Comment:
It's a python 2/3 problem that I didn't want to fiddle with :(

I am just doing str(type(obj)) and one str()'s to <class 'str'> and one does <type 'str'>.

Reply:
Okay, I have performed some skulduggery to make this work in Python 2 and 3. Because str in Python 2 isn't a class (but it is in Python 3) its ``__repr__`` differs (I didn't look at the actual source so I'm just guessing here). I made a new-style object for the purpose of these tests which is represented the same way in both Python 2 and 3.

Reply:
I was going to say don't bother with it...

Reply:
Eh, it's slightly neater now, I don't mind